patchwork-fml/src/main/java/net/minecraftforge/fml/ModList.java

```diff
@@ -33,4 +45,20 @@ public boolean isLoaded(String modId) {
 		// Patchwork: use Fabric Loader lookup instead of an internal one
 		return FabricLoader.getInstance().isModLoaded(modId);
 	}
+
+	public ModFileInfo getModFileById(String modId) {
+		return modFileInfoMap.computeIfAbsent(modId, ModFileInfo::new);
+	}
+
+	public List<ModFileScanData> getAllScanData() {
+		if (allScanDataCache == null) {
+			allScanDataCache = FabricLoader.getInstance().getAllMods()
+					.stream()
+					.map(modContainer -> modContainer.getMetadata().getId())
+					.map(modid -> getModFileById(modid).getFile().getScanResult())
```

Comment:
Won't this duplicate annotation data due to multiple mods sharing the same annotation data (jar-in-jar)? I think the root problem is that we associate one mod file with one mod ID, instead of the mod IDs within the same file sharing the same ModFileInfo like they would in Forge. We might need a way to indicate in general when a given JiJ mod is just a dummy mod for metadata purposes with something like `patchwork:parent` or `patchwork:dummy` on the patcher side.

Reply:
Changed to make jij mod and parent mod share the same ModFileInfo